store/badger.go

```diff
@@ -49,21 +53,45 @@ func (b *badgerWriteBatch) Destroy() {
 type badgerKV struct {
 	dir string
 	db  *badger.DB
+
+	// Stops gc goroutine when db is closed.
+	closeCh chan struct{}
+	closeWg sync.WaitGroup
 }
 
-func NewBadger(dir string) (*badgerKV, error) { // nolint:golint
+func NewBadger(dir string) (*badgerKV, error) { // nolint:golint,gocritic
+	if fi, err := os.Stat(dir); err == nil {
```

Comment:
The linter really wants this to be a switch statement but I think it makes more sense as written than as switch.  Is there an even more clear way to write this ?

One option might be to unconditionally call os.Mkdir() and ignore the return value, then let `badger.Open()` fail if the conditions are still wrong (as long as it returns a meaningful error message)